src/Select/index.js

```diff
@@ -58,9 +58,16 @@ export const getItemByValue = (value, items) => {
  * @param {array} items downshift index `items`
  * @returns {Number} index of item
  */
-export const getItemIndex = ({ props }, items) => {
-  const values = items.map(({ props }) => props.value);
-  return values.indexOf(props.value);
+export const getItemIndex = (item, items) => {
+  let result = 0;
+  if (isAction(item)) {
+    result = items
+      .map(({ props }) => props.onSelect)
+      .indexOf(item.props.onSelect);
+  } else {
+    result = items.map(({ props }) => props.value).indexOf(item.props.value);
+  }
+  return result;
 };
```

Comment:
These changes exposed a bug in `getItemIndex` where it fails on an `Select.Action` subcomponent.

`Select.Item` can be found by their `value` prop. `Select.Action` should be found by their `onSelect`.